compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/checkers/FirHelpers.kt

```diff
@@ -318,3 +322,9 @@ internal fun throwableClassLikeType(session: FirSession) = session.builtinTypes.
 
 fun ConeKotlinType.isSubtypeOfThrowable(session: FirSession) =
     throwableClassLikeType(session).isSupertypeOf(session.typeContext, this.fullyExpandedType(session))
+
+val FirValueParameter.hasValOrVar: Boolean
+    get() {
+        val source = this.source ?: return false
+        return source.getChild(VAL_KEYWORD) != null || source.getChild(VAR_KEYWORD) != null
```

Comment:
There is `VAL_VAR_TOKEN_SET`.

However, can we just do something like:
```kt
val FirValueParameter.isValOrVar: Boolean
    get() = isVal || isVar
```
?

Reply:
`isVal` and `isVar` don't work. I don't know how these two properties are used for a `FirValueParameter` . These values are statically set in `FirValueParameterImpl` to `true` or `false`

Reply:
Yeah, it looks like value parameter with `val` or `var` is converted to property, then remaining value parameter doesn't set `isVal` or `isVar` correctly. You can leave a TODO, like, "// TODO: better to set [FirValueParameter]'s isVal and isVar ?" and then I can take a look at it later.

@mglukhikh what do you think? Having `isVal` and `isVar` there but not set them correctly and not using them seem like wasting of those two fields.

Reply:
As discussed during the previous sync. We will keep the `isVal` and `isVar` in `KtValueParameter` as it is.